src/apps/frontend/pages/authentication/reset-password/reset-password-form.hook.ts

```diff
@@ -0,0 +1,59 @@
+import { useFormik } from 'formik';
+import { useLocation, useParams } from 'react-router-dom';
+import * as Yup from 'yup';
+
+import constant from '../../../constants';
+import { useResetPasswordContext } from '../../../contexts';
+import { AsyncError } from '../../../types';
+
+interface ResetPasswordFormProps {
+  onSuccess: () => void;
+  onError: (err: AsyncError) => void;
+}
+
+const useResetPasswordForm = ({ onError, onSuccess }: ResetPasswordFormProps) => {
+  const { accountId } = useParams();
+
+  const { search } = useLocation();
+  const token = new URLSearchParams(search).get('token');
+
+  const {
+    isResetPasswordLoading, resetPasswordError, resetPassword,
+  } = useResetPasswordContext();
+
+  const formik = useFormik({
+    initialValues: {
+      password: '',
+      retypePassword: '',
+    },
+    validationSchema: Yup.object({
+      password: Yup.string()
+        .min(constant.PASSWORD_MIN_LENGTH, constant.PASSWORD_VALIDATION_ERROR)
+        .required(constant.PASSWORD_VALIDATION_ERROR),
+      retypePassword: Yup.string()
+        .oneOf([Yup.ref('password')], constant.PASSWORD_MATCH_VALIDATION_ERROR)
+        .required(constant.PASSWORD_MATCH_VALIDATION_ERROR),
+    }),
+    onSubmit: (values) => {
+      resetPassword(
+        accountId,
+        values.password,
+        token,
```

Comment:
We should define a type so we don't have to keep passing accountId, password and token (as string) separately.